Exiled.Events/Patches/Events/Scp3114/Dancing.cs

```diff
@@ -0,0 +1,79 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="Dancing.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Events.Scp3114
+{
+    using System.Collections.Generic;
+    using System.Reflection.Emit;
+
+    using Exiled.API.Features;
+    using Exiled.API.Features.Pools;
+    using Exiled.Events.EventArgs.Scp3114;
+    using HarmonyLib;
+    using PlayerRoles.PlayableScps.Scp3114;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Patches <see cref="Scp3114Dance.ServerProcessCmd"/>
+    /// to add <see cref="Handlers.Scp3114.Dancing"/> event.
+    /// </summary>
+    [HarmonyPatch(typeof(Scp3114Dance), nameof(Scp3114Dance.ServerProcessCmd))]
+    internal class Dancing
+    {
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(instructions);
+
+            int index = newInstructions.FindIndex(x => x.opcode == OpCodes.Brfalse_S);
+
+            Label continueLabel = generator.DefineLabel();
+
+            LocalBuilder ev = generator.DeclareLocal(typeof(DancingEventArgs));
+
+            newInstructions.InsertRange(
+                index,
+                new[]
+                {
+                    // reader.ReadBool();
```

Comment:
No ?

Reply:
Just showing that first constructor argument is above (inserting right after that)

Reply:
> Just showing that first constructor argument is above (inserting right after that)

That should be say with Index so not inside the AddRange